### DISCUSSION TITLE
image not found or typ unknown

### DISCUSSION THREAD

**Discussion by Michael-7755** [dompdf/dompdf#2352] [General]
Hello Experts

I have installed dompdf with composer and it works with the exception that images are not displayed. I did a lot of research and found a lot of suggestions - but none of them worked. It's a simple logo.png file. When I output this as HTML, the image is available but not in the PDF file. The suggestions to change the configuration does not work either.

Tanks for help
Michael

**Comment by bsweeney** [comment]
Can you provide more details about what you have tried as well as some sample code?

**Comment by Michael-7755** [comment]
Thank you fore reply
The first part:
```
require_once('dompdf/vendor/autoload.php');
$options = new Options(); //I get the error message: class Options not found
//$options->setIsRemoteEnabled(true);
//$dompdf = new Dompdf($options);
use Dompdf\Dompdf;
$dompdf = new Dompdf();
// $dompdf->set_option('isHtml5ParserEnabled', true);
// $dompdf->set_option('isRemoteEnabled', true);  
```

if I say to merkur:
michael@merkur:~$ find /var/www/html/lindenimmo3/dompdf/venor -name "*config*.php"
find: ‘/var/www/html/lindenimmo3/dompdf/venor’: Datei oder Verzeichnis nicht gefunden
Datei oder Verzeichnis nicht gefunden - in english: File or directory not found
Conclusion: there is nothing to configure

the second part:
i have try this
```
$logo="<div class=\"pdflogo\"><center><img src=\"./images/logo_pdf.png\"></center></div><br>"
```
and this
```
$logo="<div class=\"pdflogo\"><center><img src=\"https://merkur.fritz.box/lindenimmo3/images/logo_pdf.png\"></center></div><br>";
```

merkur.fritz.box is my own ubuntu 20.04 server running php 7.4.
third part:

```
$dompdf->load_html($html_template);
$dompdf->set_paper("a4", 'portrait'); 
$dompdf->render();
$canvas = $dompdf->get_canvas();
//$font = Font_Metrics::get_font("Helvetica", "normal"); 
$canvas->page_text(35, 800, $foottext." Seite {PAGE_NUM} von {PAGE_COUNT}", $font, 8, array(0,0,0));
$exfile=FOLDER."expose_exid_".$expoid.".pdf";
//echo "file: ".$exfile;
$pdfoutput = $dompdf->output();
//echo "file: ".$exfile;
$fp = fopen($exfile, "a");
//echo "fp: ".$fp." ";
fwrite($fp, $pdfoutput);
fclose($fp);
//file_put_contents("file.pdf", $pdf);
//$dompdf->stream($exfile);
```

when I do the HTML output for the test

```
echo "HTML output: ".$html_template; //this output is ok with logoimage
```

**Comment by bsweeney** [comment]
To render the image from the web (e.g. `https://merkur.fritz.box/lindenimmo3/images/logo_pdf.png`) you'll need to enable remote resource loading. One way to do that is:

```
$dompdf = new Dompdf(["isRemoteEnabled" => true]);
```

To render the image from your local filesystem (e.g. `./images/logo_pdf.png`) you'll need to ensure that it's under the path specified by chroot. So:

```
$dompdf = new Dompdf(["chroot" => __DIR__]);
```

Plus, the path has to be correct relative to the PHP script.

There are other nuances that could be interfering with your ability to render an image. For example:
- Is the GD extension enabled (required for PNG images)?
- Does your script have read/write access to the temp directory? You can specify a new directory using the "tempDir" option.
- For web-based content is the PHP setting allow_url_fopen set to true? If not, is the curl extension enabled?

Recommend you read through the wiki: https://github.com/dompdf/dompdf/wiki/Requirements

**Comment by Michael-7755** [comment]
Thank you very much - it works. 
Michael

**Reply by catumbaco** [reply]
A MI NO SE ME MUESTRA LA IMAGEN NI TAMPOCO EL DISEÑO DE LAS TABLAS, AYUDA POR FAVOR
